ipfx/bin/plot_ephys_nwb.py

```diff
@@ -0,0 +1,117 @@
+import sys
+import matplotlib.pyplot as plt
+import numpy as np
+from ipfx.data_set_utils import create_data_set
+
+DEFAULT_NWB_FILE_NAME ="/allen/programs/celltypes/workgroups/279/MIES Testing Spring 2018/Sst-IRES-CreAi14-395722.01.01.01.nwb"
+
+
+def plot_data_set(data_set, sweep_table, nwb_file_name):
+
+    height_ratios, width_ratios = axes_ratios(sweep_table)
+
+    fig, ax = plt.subplots(len(height_ratios), 3,
+                           figsize=(15, 7),
+                           gridspec_kw={'height_ratios': height_ratios, 'width_ratios': width_ratios}
+                           )
+
+    i_ax = 0
+    for stimulus_code, sweep_set_table in sweep_table.groupby("stimulus_code"):
+        sweep_numbers = sweep_set_table["sweep_number"].sort_values().values
+        ss = data_set.sweep_set(sweep_numbers)
+
+        ax_a = ax[i_ax,0]
+        ax_i = ax[i_ax,1]
+        ax_v = ax[i_ax,2]
+
+        plot_waveforms(ax_i, ss.i, ss.sampling_rate, ss.sweep_number)
+        plot_waveforms(ax_v, ss.v, ss.sampling_rate, ss.sweep_number)
+        ax_v.get_shared_x_axes().join(ax_i, ax_v)
+
+        clamp_mode = sweep_set_table["clamp_mode"].values[0]
+        ax_a.text(0, 0.0, "%s, %s " % (stimulus_code, clamp_mode))
+        ax_a.axis('off')
+
+        i_ax += 1
+
+    ax[0,0].set_title("Description")
+    ax[0,1].set_title("Current")
+    ax[0,2].set_title("Voltage")
+    ax[-1,1].set_xlabel("time (s)")
+    ax[-1,2].set_xlabel("time (s)")
+
+    fig.suptitle("file: " + nwb_file_name, fontsize=12)
+
+    mng = plt.get_current_fig_manager()
+    mng.resize(*mng.window.maxsize())
+    plt.subplots_adjust(left=0.01, right=0.98, bottom=0.02,top=0.92)
+
+
+def axes_ratios(sweep_table):
+
+    height_ratios = []
+    width_ratios = [1,3,3]
+
+    for name, sweep_set_table in sweep_table.groupby("stimulus_code"):
```

Comment:
If not using the `name` variable, I would set it to `_`. Or maybe better yet change your `sweep_table.groupby("stimulus_code")` so that it only contains `sweep_set_table`.

Reply:
thanks.